usr/share/rear/lib/layout-functions.sh

```diff
@@ -992,4 +992,251 @@ function apply_layout_mappings() {
+        last_partition_number=$number
+        return 0
+    fi
+
+    if is_false $FEATURE_PARTED_RESIZEPART && is_false $FEATURE_PARTED_RESIZE ; then
+        Error "Disk '$disk': trying to create partition number $number which isn't consecutive with previous partition but 'parted' doesn't support this feature"
+    fi
+
+    # "parted" is only capable of creating partitions consecutively. Since
+    # there is a gap between the previous partition and this partition, dummy
+    # partitions must be created, and later will be removed once disks are
+    # fully partitioned (this is done in
+    # delete_dummy_partitions_and_resize_real_ones()).
+    #
+    # Once the dummy partitions are removed, the real partition needs to be
+    # resized to original size.
+    # Unfortunately, "parted" is only capable of resizing a partition toward
+    # its end, not the beginning, so the only way to create dummy partitions
+    # appropriately is to use some space at the end of the real partition.
+    #
+    # Example: the GPT disk contains only 1 partition numbered 3 named PART
+    #
+    # # parted -m -s /dev/sda unit B print
+    # /dev/sda:21474836480B:scsi:512:512:gpt:QEMU QEMU HARDDISK:;
+    # 3:1048576B:2097151B:1048576B::PART:;
+    #
+    # We need to create these partitions to recreate the exact same partitions:
+    #
+    # # parted -m -s /dev/sda unit B print
+    # /dev/sda:21474836480B:scsi:512:512:gpt:QEMU QEMU HARDDISK:;
+    # 3:1048576B:2096127B:1047552B::PART:;
+    # 2:2096128B:2096639B:512B::dummy2:;
+    # 1:2096640B:2097151B:512B::dummy1:;
+    #
+    # Then delete dummy partition 1 and 2 and resize 3 to its original end.
+    #
+    # For 'msdos' disks, only non-logical partitions are affected.
+
+    if [[ ! $endB ]] ; then
+        # We need to compute '$endB' based on disk size. To do so, create a
+        # partition then get its end
+        LogPrint "Disk '$disk': creating a temporary partition to find out the end of the disk"
+        parted -s -m $disk mkpart "$name" ${startB}B 100%
+        local num
+        read num endB <<< $( parted -s -m $disk unit B print | tail -1 | awk -F ':' '{ print $1, $3 }' | sed 's/\([0-9]*\)B$/\1/' )
+        LogPrint "Disk '$disk': last allocatable byte on disk is '$endB'"
+        LogPrint "Disk '$disk': deleting the temporary partition number $num"
+        parted -s -m $disk rm $num
+    fi
+
+    partitions_to_resize+=( $number $endB )
+
+    local -i logical_sector_size=$( parted -m -s $disk unit B print | awk -F ':' "\$1 == \"$disk\" { print \$4 }" )
+
+    if [[ "$disk_label" != "msdos" ]] || [[ "$name" != "logical" ]] ; then
+        local -i i=$last_partition_number+1
+        while [[ $i -lt $number ]] ; do
+            local partname="dummy$i"
+            [[ "$disk_label" != "msdos" ]] || partname="primary"
+            dummy_partitions_to_delete+=( $i )
+            LogPrint "Disk '$disk': creating partition number $i with name '$partname' (will be deleted later)"
```

Comment:
Could you change that LogPrint message to
```
LogPrint "Disk '$disk': creating dummy partition number $i with name '$partname' (will be deleted later)"
```
so that it matches better to its counterpart message
in delete_dummy_partitions_and_resize_real_ones()
```
LogPrint "Disk '$current_disk': deleting dummy partition number $num"
```